test/mock-allocator.h

```diff
@@ -57,4 +57,24 @@ template <typename Allocator> class allocator_ref {
   void deallocate(value_type* p, size_t n) { alloc_->deallocate(p, n); }
 };
 
+template <typename Allocator, size_t MaxSize>
+class allocator_max_size: public Allocator {
+ public:
+  typedef typename Allocator::value_type value_type;
+  size_t max_size() const FMT_NOEXCEPT {
+    return MaxSize;
+  }
+  value_type* allocate(size_t n) {
+    if (n > max_size()) {
+      throw std::length_error("size > max_size");
+    }
+    return std::allocator_traits<Allocator>::allocate(
+      *static_cast<Allocator *>(this), n);
+  }
+  void deallocate(value_type* p, size_t n) {
+    std::allocator_traits<Allocator>::deallocate(
+      *static_cast<Allocator *>(this), p, n);
+  }
+};
```

Comment:
Please move this to the `format-test.cc` since it's only used there and not a mock.